### PR TITLE
Password reset and account editing

### PR DESCRIPTION
**Pull request by brandonson**

### CODE REVIEW

File: public/js/editAcct/editAcct.controller.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+(function () {
+	'use strict'
+
+	angular.module('editAcct')
+		.controller('editAcctCtrl', function ($scope, $state, ajax) {
+			$scope.showError = false
+			$scope.showChangeMessage = false
+			$scope.errorMessage = ""
+
```
Comment (movingpylon) [inline public/js/editAcct/editAcct.controller.js]: Need to initialize $scope.account to empty object here

File: public/js/editAcct/editAcct.controller.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+				ajax.serviceCall('Setting data...', 'post', '/api/editAccount', obj).then(function(resData) {
+					console.log("showing msg")
+					$scope.showChangeMessage = true
+				},
+				function (resData) {
+					$scope.errorMessage = "Failed to change account data"
+					$scope.showError = true
+				})
```
Comment (movingpylon) [inline public/js/editAcct/editAcct.controller.js]: This should go within the if statement

**Comment by brandonson** [comment]
When you say the password reset doesn't seem to work, do you mean you don't get an email? Or is it something else?

Not getting email would be expected, since it requires config you probably won't have set up.

**Comment by movingpylon** [comment]
I get a 500 error

**Comment by brandonson** [comment]
Actually that might be what happens when email send fails.  I'll double check later and configure the thetis server to properly send, then it can get tested.

File: public/js/editAcct/editAcct.controller.js (hunk 2 above)
Comment (brandonson) [inline public/js/editAcct/editAcct.controller.js]: I actually don't think we want it to - this way, we edit the receiveUpdates field appropriately even if the password isn't changed, unless the password is changed but invalid.

Comment (movingpylon) [inline public/js/editAcct/editAcct.controller.js]: Ok but you should probably check if the password is an empty string

**Comment by nikkimartin** [comment]
mentioned in slack but also putting in here just in case:
- link to edit account page instead of telling users to go their in email instructions
- include conference name in event update email (e.g. "The event PartyNight for conference Foo has been updated.")

File: public/js/editAcct/editAcct.controller.js (hunk 2 above)
Comment (brandonson) [inline public/js/editAcct/editAcct.controller.js]: Pretty sure it checks that - the if statement checks `account.password && account.password.length >= 8`

In any case, when I try to set an empty password, it just ignores the password and only sets `receiveUpdates`, which should be fine.

**Comment by brandonson** [comment]
Got the second part of @nikkimartin's comment, linking to edit account page doesn't work nicely due to how we reject unauthed users and so won't happen.

**Pull request merged**